SimGeneral/PreMixingModule/python/mixOne_premix_on_sim_cfi.py

```diff
@@ -0,0 +1,252 @@
+# This is the PreMixing config. Not only does it do a RawToDigi
+# conversion to the secondary input source, it also holds its own
+# instances of an EcalDigiProducer and an HcalDigitizer. It also
+# replicates the noise adding functions in the SiStripDigitizer.
+#
+# Adapted from DataMixingModule
+
+
+import FWCore.ParameterSet.Config as cms
+from SimCalorimetry.HcalSimProducers.hcalUnsuppressedDigis_cfi import hcalSimBlock
+from SimGeneral.MixingModule.SiStripSimParameters_cfi import SiStripSimBlock
+from SimGeneral.MixingModule.SiPixelSimParameters_cfi import SiPixelSimBlock
+from SimGeneral.MixingModule.ecalDigitizer_cfi import ecalDigitizer
+
+import EventFilter.EcalRawToDigi.EcalUnpackerData_cfi
+import EventFilter.ESRawToDigi.esRawToDigi_cfi
+import EventFilter.HcalRawToDigi.HcalRawToDigi_cfi
+import EventFilter.DTRawToDigi.dtunpacker_cfi
+import EventFilter.RPCRawToDigi.rpcUnpacker_cfi
+import EventFilter.CSCRawToDigi.cscUnpacker_cfi
+import EventFilter.SiStripRawToDigi.SiStripDigis_cfi
+import EventFilter.SiPixelRawToDigi.SiPixelRawToDigi_cfi
+
+# content from Configuration/StandardSequences/DigiToRaw_cff.py
+
+ecalDigis = EventFilter.EcalRawToDigi.EcalUnpackerData_cfi.ecalEBunpacker.clone()
+
+ecalPreshowerDigis = EventFilter.ESRawToDigi.esRawToDigi_cfi.esRawToDigi.clone()
+
+hcalDigis = EventFilter.HcalRawToDigi.HcalRawToDigi_cfi.hcalDigis.clone()
+
+muonCSCDigis = EventFilter.CSCRawToDigi.cscUnpacker_cfi.muonCSCDigis.clone()
+
+muonDTDigis = EventFilter.DTRawToDigi.dtunpacker_cfi.muonDTDigis.clone()
+
+siStripDigis = EventFilter.SiStripRawToDigi.SiStripDigis_cfi.siStripDigis.clone()
+
+siPixelDigis = EventFilter.SiPixelRawToDigi.SiPixelRawToDigi_cfi.siPixelDigis.clone()
+
+siPixelDigis.InputLabel = 'rawDataCollector'
+ecalDigis.InputLabel = 'rawDataCollector'
+ecalPreshowerDigis.sourceTag = 'rawDataCollector'
+hcalDigis.InputLabel = 'rawDataCollector'
+muonCSCDigis.InputObjects = 'rawDataCollector'
+muonDTDigis.inputLabel = 'rawDataCollector'
+
+hcalDigis.FilterDataQuality = cms.bool(False)
+hcalSimBlock.HcalPreMixStage2 = cms.bool(True)
+
+mixData = cms.EDProducer("PreMixingModule",
```

Comment:
Maybe we should consider if we still want this module to be named `mixData`?

Reply:
Eventually we may want to change the name, but at the moment too many things depend on the `mixData` name (including `ConfigBuilder`) that in this PR it will stay.